### DISCUSSION TITLE
Prepared Statements and Performance Schema

### DISCUSSION THREAD

**Discussion by ganderberg-akoustis** [mkleehammer/pyodbc#1227] [Q&A] [answered]
I am using pyODBC as the library I connect to a server (MariaDB 10.6) but I am having issues with the current setup of the pyODBC library. When performance schema tracks the statement events I send with the library, it does not track the actual statement. See shown below:

![image](https://github.com/mkleehammer/pyodbc/assets/135854921/a4fa1b88-a7f2-49e9-b1fe-b156fa8048c6)

This makes sense, since the pyODBC library states that it uses prepared statements for performance with repeated statements. The issue is I can not differentiate between these evens because they only list as 'statement/com/Prepare', 'statement/com/Execute'.

My question is, is it possible to have a cursor or connection execute the SQL statement without going through the prepared statement? Is there an underlying function I can use that interacts with the server directly? Is there an option to disable the prepared statement setup?

**Comment by gordthompson** [comment]
Have you tried adding `PREPONCLIENT=1` to your connection string?

https://mariadb.com/kb/en/about-mariadb-connector-odbc/#parameters

**Reply by ganderberg-akoustis** [reply, marked as answer]
Updated to MariaDB 3.2.0 Connector and this works great! Many thanks!
